Remotion/Core/Core/Collections/DictionaryExtensions.cs

```diff
@@ -79,11 +86,13 @@ public static class DictionaryExtensions
         return defaultValue;
     }
 
+    [return: MaybeNull]
     public static TValue GetValueOrDefault<TKey, TValue> (this Dictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue)
     {
       return ((IDictionary<TKey, TValue>) dictionary).GetValueOrDefault (key, defaultValue);
     }
 
+    [return: MaybeNull]

```

Comment:
since its used as an overload with `default` i think it should probably be 
```csharp
[return: MaybeNull, NotNullIfNotNull("defaultValue")]
public static TValue GetValueOrDefault<TKey, TValue> (this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key, [AllowNull] TValue defaultValue)
```

Reply:
It is, but the default value is passed in as `TSource defaultvalue`. If the dictionary is has nullable values, then the default can be null - if not, the user must think about a sensible default.

Apart from the desired intent of use, `NotNullIfNotNull` seems to "override" `MaybeNull` - if `defaultValue` is `TSource!`, no null check needs to be performed upon usage of the return value. Did you have other observations/thoughts?

Reply:
With the same reasoning, I would remove `AllowNull` from `TSource defaultValue`.

Reply:
If you want, you could also leave it as-is for MKs review, please leave this discussion open.